src/main/java/sirius/biz/jobs/batch/file/EntityImportSyncJob.java

```diff
@@ -0,0 +1,234 @@
+                               ProcessContext process,
+                               String factoryName) {
+        super(fileParameter, ignoreEmptyParameter, dictionary, process);
+        this.importer.setFactoryName(factoryName);
+        this.importTransactionHelper = importer.findHelper(ImportTransactionHelper.class);
+        this.mode = process.getParameter(syncModeParameter).orElse(SyncMode.NEW_AND_UPDATE_ONLY);
+        this.type = type;
+        this.descriptor = mixing.getDescriptor(type);
+    }
+
+    /**
+     * Specifies a context extender which can be used to transfer job parameters into the import context.
+     *
+     * @param contextExtender the extender to specify
+     * @return the import job itself for fluent method calls
+     */
+    public EntityImportSyncJob<E, Q> withContextExtender(Consumer<Context> contextExtender) {
+        this.contextExtender = contextExtender;
+        return this;
+    }
+
+    public EntityImportSyncJob<E, Q> withDeleteQueryTuner(BiConsumer<ProcessContext, Q> queryTuner) {
+        this.queryTuner = queryTuner;
+        return this;
+    }
+
+    @Override
+    protected void executeForStream(String filename, InputStream in) throws Exception {
+        importTransactionHelper.start();
+        LineBasedProcessor.create(filename, in).run(this, error -> {
+            process.handle(error);
+            return true;
+        });
+        commitImportTransaction();
+    }
+
+    /**
+     * Commits the import transaction by deleting all untouched entities.
+     */
+    protected void commitImportTransaction() {
+        if (mode != SyncMode.SYNC) {
+            return;
+        }
+
+        Watch watch = Watch.start();
+        importTransactionHelper.deleteUnmarked(type, this::tuneImportTransactionDeleteQuery, entity -> {
+            process.addTiming(NLS.get("EntityImportJob.entityDeleted"), watch.elapsed(TimeUnit.MILLISECONDS, true));
+        });
+    }
+
+    /**
+     * Tunes the delete query of the import transaction so that all untouched entities will be deleted.
+     *
+     * @param deleteQuery the query to enhance
+     */
+    protected void tuneImportTransactionDeleteQuery(Q deleteQuery) {
+        queryTuner.accept(process, deleteQuery);
+    }
+
+    @Override
+    protected final void handleRow(int index, Context context) {
+        Watch watch = Watch.start();
+
+        if (contextExtender != null) {
+            contextExtender.accept(context);
+        }
+
+        E entity = findAndLoad(context);
+        if (mode == SyncMode.DELETE_EXISTING) {
+            if (!entity.isNew()) {
+                importer.deleteNow(entity);
+                process.addTiming(NLS.get("EntityImportJob.entityDeleted"), watch.elapsedMillis());
+            }
+        } else {
+            createorUpdateEntity(entity, context, watch);
+        }
+    }
+
+    private void createorUpdateEntity(E entity, Context context, Watch watch) {
```

Comment:
```suggestion
    private void createOrUpdateEntity(E entity, Context context, Watch watch) {
```
und halt an den anderen stellen auch